spec/acceptance/api/service_spec.rb

```diff
@@ -40,7 +53,11 @@
 
     context 'service' do
       subject(:service) { Hash.from_xml(serialized).fetch(root) }
-      it { should include('id' => resource.id.to_s, 'system_name' => resource.system_name) }
+      it { should include(attributes.map do |attr_name|
+        next if (attr_value = resource.public_send(attr_name)).blank?
```

Comment:
You don't need this line having the compact later, no? It should work like this:
```ruby
it { should include(attributes.map { |attr_name| [attr_name, resource.public_send(attr_name).to_s.presence] }.compact.to_h) }
```

Reply:
Yes and no. If we just collect the pairs without skipping when blank and without `compact`, it will work for this test in particular. However it's more about communicating through what's expected. If an attribute of `resource` is blank, it won't be included. I know, I should have added another test, but it was never there even though it's been the behavior expected since ever.